pkg/dockerregistry/server/manifestservice.go

```diff
@@ -42,6 +42,10 @@ func (m *manifestService) Exists(ctx context.Context, dgst digest.Digest) (bool,
 
 	image, err := m.imageStream.GetImageOfImageStream(ctx, dgst)
 	if err != nil {
+		switch err.Code {
+		case imagestream.ErrImageStreamNotFoundCode, imagestream.ErrImageStreamImageNotFoundCode:
+			return false, distribution.ErrBlobUnknown
```

Comment:
why don't we do the same thing here as we do for Get?

Reply:
Because the `manifestService.Get` should return only `distribution.ErrManifestUnknownRevision` [1] and the `manifestService.Exists` should return `distribution.ErrBlobUnknown` [2].

[1] https://github.com/legionus/image-registry/blob/250164d82a9ab20d7e580cdfb3866766c2536551/vendor/github.com/docker/distribution/registry/handlers/images.go#L100-L108
[2] https://github.com/legionus/image-registry/blob/250164d82a9ab20d7e580cdfb3866766c2536551/vendor/github.com/docker/distribution/registry/storage/manifestlisthandler.go#L77